mlptrain/configurations/configuration_set.py

```diff
@@ -617,8 +617,10 @@ def _run_parallel_method(self, function, **kwargs):
                                           kwds=kwargs)
                 results.append(result)
 
+            pool.close()
             for i, result in enumerate(results):
                 self[i] = result.get(timeout=None)
+            pool.join()
```

Comment:
I didn't think you needed to do this with a mp pool 😕 . doc: https://docs.python.org/3/library/multiprocessing.html#using-a-pool-of-workers

have you got a test that fails without this and succeeds with this?

Reply:
We are sometimes getting errors when using linear ACE and we are not entirely sure why. Most of them arise when performing tasks in parallel, and we've found that adding pool.close() and pool.join() relieves some of these errors.